### PR TITLE
Ajax track pull

### PR DESCRIPTION
**Pull request by torpeyp**
@liammccartney could you just take a peek at this to see what can be done about that test. I've out of it now and I've wasted to much time trying to find a roundabout. I'm going to get something to eat or something.

### CODE REVIEW

**Comment by liammccartney** [comment]
@torpeyp - not sure that this is good to merge, let's circle back on it in the morning.

File: app/assets/javascripts/soundcloud_user.js [hunk 1]
```diff
@@ -11,9 +11,9 @@ SoundcloudUser.prototype.searchSongs = function(){
         url: '/tracks',
         method: 'POST',
         data: {tracks: foundTracks},
-        dataType: 'JSON'
-      }).success(function(response){
-        $('body').append(response)
+      }).done(function(response){
+        $('.user-tracks').append(response);
+      }).fail(function(){
```
Comment (sgharms) [inline app/assets/javascripts/soundcloud_user.js]: kim crying

Dude, an empty `fail()` function.  Even `alert('bad news')` would be great.

File: app/controllers/tracks_controller.rb [hunk 2]
```diff
@@ -13,11 +13,11 @@ def new
   end
 
   def create
-
     respond_to do |format|
```
Comment (sgharms) [inline app/controllers/tracks_controller.rb]: This thing is ugly.  AGguughhahghh.

File: app/controllers/tracks_controller.rb [hunk 3]
```diff
@@ -26,9 +26,10 @@ def create
           waveform_url: track_data['waveform_url'],
           bpm: track_data['bpm']
           )
-        next unless track.save
-      end
-      render plain: 'AJAX OK'
+          next unless track.save
+          tracks << track
```
Comment (sgharms) [inline app/controllers/tracks_controller.rb]: Oh i think i may know what's bothering me, are you using `#each` to accumulate to an array?  That's what `#map` is for.

File: app/controllers/tracks_controller.rb [hunk 4]
```diff
@@ -40,6 +41,7 @@ def create
         end
       }
     end
+
```
Comment (sgharms) [inline app/controllers/tracks_controller.rb]: don't add blank whitespace plz

File: app/models/track.rb [hunk 5]
```diff
@@ -3,6 +3,13 @@ class Track < ActiveRecord::Base
   belongs_to :user
   has_many :votes, as: :votable
 
+  def string_duration
```
Comment (sgharms) [inline app/models/track.rb]: why not `#to_s` ?

**Comment by sgharms** [comment]
@liammccartney @torpeyp take a look at these.  it's a good PR, but there's some slop :tm: in there.  Let's be focused here.

**Pull request closed without merging**